src/blocks/weather/met_no.rs

```diff
@@ -0,0 +1,221 @@
+use super::*;
+
+type LegendsStore = HashMap<String, LegendsResult>;
+
+#[derive(Deserialize)]
+#[serde(tag = "name", rename_all = "lowercase")]
+pub(super) struct Config {
+    coordinates: Option<(String, String)>,
+    altitude: Option<String>,
+    #[serde(default)]
+    lang: ApiLanguage,
+}
+
+pub(super) struct Service {
+    config: Config,
+    legend: LegendsStore,
+}
+
+impl Service {
+    pub(super) async fn new(config: Config) -> Result<Self> {
+        Ok(Self {
+            config,
+            legend: get_legend().await?,
```

Comment:
I think this should be inside of `api.recoverable` so that if the bar is launched before computer gains internet access, the block will eventually get working. So I suppose this `new` function should also accept `api: &mut CommonApi`.

Reply:
Fixed in b216878.